gui/app/(tabs)/registration.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useState } from 'react';
+import { View, Text, TextInput, StyleSheet, TouchableOpacity, Image } from 'react-native';
+import * as ImagePicker from 'expo-image-picker';
+
+interface RegistrationFormData {
+    firstName: string;
+    lastName: string;
+    username: string;
+    email: string;
+    password: string;
+    profilePicture: string | null;
+}
+
+const RegistrationScreen: React.FC = () => {
+    const [formData, setFormData] = useState<RegistrationFormData>({
+        firstName: '',
+        lastName: '',
+        username: '',
+        email: '',
+        password: '',
+        profilePicture: null,
+    });
+
+    const handleChoosePhoto = async () => {
+        const { status } = await ImagePicker.requestMediaLibraryPermissionsAsync();
+        if (status !== 'granted') {
+            alert('Sorry, we need camera roll permissions to make this work!');
+            return;
+        }
+
+        const result = await ImagePicker.launchImageLibraryAsync({
+            mediaTypes: ImagePicker.MediaTypeOptions.Images,
+            allowsEditing: true,
+            aspect: [4, 4],
+            quality: 1,
+        });
+
+        if (!result.canceled) {
+            setFormData({ ...formData, profilePicture: result.assets[0].uri });
+        }
+    };
+
+    const handleFirstNameChange = (text: string) => {
+        setFormData({ ...formData, firstName: text });
+    };
+
+    const handleLastNameChange = (text: string) => {
+        setFormData({ ...formData, lastName: text });
+    };
+
+    const handleUserNameChange = (text: string) => {
+        setFormData({ ...formData, username: text });
+    };
```

Comment:
Can probably be condensed to just a handleNameChange() method.

Are we going to have the functionality for someone to change their name? It might be worth a shot to put these functions outside of the registration page logically, either into a util or just contained in their own component. That way the registration page and some editProfile page can pull in the same component and use the same stuff.

Also, any way to add some quick validation to the form? i.e. make sure the password is the right length of the name doesn't contain any invalid characters?